uaclient/data_types.py

```diff
@@ -122,6 +123,21 @@ def from_value(val: Any) -> bool:
         return val
 
 
+class DatetimeDataValue(DataValue):
+    """
+    This expects that value is a datetime.
+    Use util.DatetimeAwareJSONDecoder when parsing json
```

Comment:
Sorry @orndorffgrant, I think I am missing the point of this part of the docstring.

Should we also update the `DataObject` `to_json` method to use the `DatetimeAwareJSONDecoder` class ?

Reply:
Thanks @lucasmoura. I added the DatetimeAwareJSONEncoder to the `to_json` method and corresponding tests. And I removed the confusing line in the docstring - it didn't really add anything anyway